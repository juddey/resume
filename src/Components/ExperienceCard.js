import React from 'react'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import Avatar from '@material-ui/core/Avatar'
import IconButton from '@material-ui/core/IconButton'
import Chip from '@material-ui/core/Chip'
import { ChevronDownIcon, ChevronUpIcon } from 'mdi-react'
import { Images } from '../Themes'

class ExperienceCard extends React.Component {
  constructor (props) {
    super(props)
    this.state = { expanded: false }
  }

  handleExpandClick () {
    this.setState({ expanded: !this.state.expanded })
  }

  render () {
    return (
      <div style={{ minWidth: 275 }}>
        <Card>
          <CardHeader
            avatar={
              <Avatar aria-label='tiaki' alt='tiaki' src={Images.tiaki} />
            }
            title={
              <div
                style={{
                  display: 'flex',
                  flex: 1,
                  flexDirection: 'row'
                }}
              >
                <div
                  style={{
                    display: 'flex',
                    flexDirection: 'row',
                    flex: 1
                  }}
                >
                  <div
                    style={{
                      display: 'flex',
                      flexDirection: 'row',
                      alignItems: 'center',
                      flex: 1,
                      justifyContent: 'space-between',
                      marginRight: 4
                    }}
                  >
                    <div>Founder, tiaki.care</div>
                    <div>2015 - now</div>
                  </div>
                  <IconButton onClick={this.handleExpandClick.bind(this)}>
                    {this.state.expanded ? (
                      <ChevronUpIcon />
                    ) : (
                      <ChevronDownIcon />
                    )}
                  </IconButton>
                </div>
              </div>
            }
          />
          <div>
            {this.state.expanded && (
              <div style={{ marginTop: -20 }}>
                <CardContent>
                  <Typography paragraph>
                    creating a SaaS for early childhood centres
                  </Typography>

                  <Chip
                    label={'React Native'}
                    style={{
                      marginTop: 5,
                      background: 'white',
                      marginRight: 5,
                      border: '1px solid black'
                    }}
                  />
                  <Chip
                    label={'React'}
                    style={{
                      marginTop: 5,
                      background: 'white',
                      marginRight: 5,
                      border: '1px solid black'
                    }}
                  />
                  <Chip
                    label={'Elixir'}
                    style={{
                      marginTop: 5,
                      background: 'white',
                      marginRight: 5,
                      border: '1px solid black'
                    }}
                  />
                  <Chip
                    label={'Ruby'}
                    style={{
                      marginTop: 5,
                      background: 'white',
                      marginRight: 5,
                      border: '1px solid black'
                    }}
                  />
                  <Chip
                    label={'AWS'}
                    style={{
                      marginTop: 5,
                      background: 'white',
                      marginRight: 5,
                      border: '1px solid black'
                    }}
                  />
                </CardContent>
              </div>
            )}
          </div>
        </Card>
      </div>
    )
  }
}

export default ExperienceCard
