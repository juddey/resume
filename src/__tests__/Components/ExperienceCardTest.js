import React from 'react'
import {
  createComponentWithIntl,
  shallowWithIntl
} from '../../../src/Support/Utils'
import ExperienceCard from '../../Components/ExperienceCard'

const div = document.createElement('div')
const wrapper = shallowWithIntl(<ExperienceCard />)

test('component exists', () => {
  expect(wrapper.length).toBe(1)
})

test('component structure', () => {
  expect(wrapper.name()).toEqual('div')
  expect(wrapper.children().length).toEqual(1)
  expect(
    wrapper
      .children()
      .first()
      .name()
  ).toEqual('div')
  expect(
    wrapper
      .children()
      .first()
      .containsMatchingElement('ExperienceCard Component')
  ).toEqual(true)
})('renders to snapshot', () => {
  const tree = createComponentWithIntl(<ExperienceCard />, div).toJSON()
  expect(tree).toMatchSnapshot()
})

// test('should do some other things', () => {
// const wrapper = shallow(<ExperienceCard // SomeProps > )
//   You can add in props as shown above, or use the constant wrapper declared
//   at the top of the file.
// })
