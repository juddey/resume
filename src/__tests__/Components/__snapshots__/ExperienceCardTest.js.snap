// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to snapshot 1`] = `
<div
  style={
    Object {
      "minWidth": 275,
    }
  }
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1"
  >
    <div
      className="MuiCardHeader-root-29"
    >
      <div
        className="MuiCardHeader-avatar-30"
      >
        <div
          aria-label="tiaki"
          className="MuiAvatar-root-35"
        >
          <img
            alt="tiaki"
            className="MuiAvatar-img-37"
            sizes={undefined}
            src="tiaki-man.png"
            srcSet={undefined}
          />
        </div>
      </div>
      <div
        className="MuiCardHeader-content-32"
      >
        <span
          className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
        >
          <div
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginRight": 4,
                  }
                }
              >
                <div>
                  Founder, tiaki.care
                </div>
                <div>
                  2015 - now
                </div>
              </div>
              <button
                className="MuiButtonBase-root-68 MuiIconButton-root-62"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-67"
                >
                  <svg
                    className="mdi-icon "
                    fill="#000"
                    height={24}
                    viewBox="0 0 24 24"
                    width={24}
                  >
                    <path
                      d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-71"
                />
              </button>
            </div>
          </div>
        </span>
      </div>
    </div>
    <div />
  </div>
</div>
`;
